### PR TITLE
entry: allow empty titles for date-only journals

### PR DESCRIPTION
**Pull request by kotachisam**
Relaxes empty-title validation so users can create entries identified purely by date — useful for
  daily-journalling workflows where the date is already the primary identifier and a title is redundant.

  ## Why

  For date-driven journalling (e.g. a Notion-like "🔏 DAILY Journal" pattern), typing a placeholder title
  for every entry is friction. The entry model already stores a date, and the list view already shows it.

  ## Behaviour

  - **Before:** entry popup blocks confirmation with "Title can't be empty."
  - **After:** empty title is accepted. Journal list renders the date (and any tags) without the blank
  title line.

  ## Scope

  - `src/app/ui/entry_popup/mod.rs` — `validate_title` is now a no-op (function kept for future validation
  like max-length).
  - `src/app/ui/entries_list/mod.rs` — skip title rendering when title is empty or whitespace, so
  `textwrap::wrap("")` edge cases don't produce a blank line.

  ## Testing

  - `cargo check` / `cargo clippy --all-targets --all-features` — clean, no new warnings.
  - `cargo test --all-features` — all existing tests pass.
  - Manual smoke: created an entry with empty title, confirmed the popup accepts it, and verified the list
  shows date+tags with no title line artifact.

  ## Out of scope

  The related "auto-fill title from date when empty" behaviour (another reasonable UX choice) is
  deliberately not included — this PR just removes the block. Happy to iterate if you'd prefer a different
  default presentation.  Journalling workflows that key off the entry date (e.g. daily
  entries named purely by date) previously had to type a placeholder
  title to satisfy validation — "Title can't be empty" blocked the
  entry popup from confirming. Relax this so the title field is
  optional.

  - entry_popup::validate_title no longer rejects empty input; it just
    clears any previous error state. Function shape preserved so future
    validation (e.g. max length) can be reinstated without restructuring.
  - entries_list renders skip the title line entirely when the title is
    empty/whitespace, so the date (and any tags) remain visible cleanly
    without an awkward blank row.

  Verified with cargo check, cargo clippy --all-targets --all-features
  and the full test suite (no pre-existing tests for validate_title; no
  new regressions).

### CODE REVIEW

File: src/app/ui/entry_popup/mod.rs [hunk 1]
```diff
@@ -339,11 +339,7 @@ impl EntryPopup<'_> {
     }
 
     fn validate_title(&mut self) {
-        if self.title_txt.lines()[0].is_empty() {
-            self.title_err_msg = "Title can't be empty".into();
-        } else {
-            self.title_err_msg.clear();
-        }
+        self.title_err_msg.clear();
```
Comment (AmmarAbouZor) [inline src/app/ui/entry_popup/mod.rs:342]: I can't believe this is reviewed by developer after the AI gave the cheapest change.

Please be aware that submitting code like this will have consequences in the future with the new [AI Policiy](https://github.com/AmmarAbouZor/tui-journal#contribution-and-ai-policy)

**Review by AmmarAbouZor — COMMENTED** [review]
I don't think this is a good solution for what you want to achieve.
Data sources expect to have titles and this could break them in ways that we didn't count for.
Please open an issue with what you want to achieve so we can discuss it and find a solution.
Also please read the new policy for AI generated contributions to avoid getting banned

**Comment by AmmarAbouZor** [comment]
I don't see this as a good fit for the current architecture and it would break things most likely.
Please open an issue to find better solutions to your use-case.

Also if you don't agree with the directions of this project, please feel free to maintain your own fork. Keeping forks in sync is really simple nowadays with the help of AI agents

**Comment by kotachisam** [comment]
@AmmarAbouZor fair point on the data-source implications - I hadn't thought through the downstream effects. Appreciate you closing this and if you like, I can open an issue to discuss the use case. Apologies for not following the issue-first process.
